recipes/musl/all/conanfile.py

```diff
@@ -0,0 +1,92 @@
+    license = "MIT"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "fPIC": [True, False],
+        "shared": [True, False]
+    }
+    default_options = {
+        "fPIC": True,
+        "shared": False
+    }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("musl is only supported on Linux.")
+        if str(self.settings.compiler) not in ["gcc", "clang"]:
+            raise ConanInvalidConfiguration("musl is only supported with GCC and Clang.")
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = self.name + "-" + self.version
+        os.rename(extracted_folder, self._source_subfolder)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        conf_args = []
+        if self.settings.compiler == "gcc":
+            conf_args.append("--enable-wrapper=gcc")
+        elif self.settings.compiler == "clang":
+            conf_args.append("--enable-wrapper=clang")
+        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        autotools = self._configure_autotools()
+        autotools.make()
+
+    def package(self):
+        self.copy("COPYRIGHT", dst="licenses", src=self._source_subfolder)
+        autotools = self._configure_autotools()
+        autotools.install()
+
+    def package_info(self):
+        bindir = os.path.join(self.package_folder, "bin")
+        self.output.info('Appending PATH environment variable: {}'.format(bindir))
+        self.env_info.PATH.append(bindir)
+
+        musl_cc = None
+        musl_cxx = None
+        musl_ld = None
+
+        if self.settings.compiler == "gcc":
+            musl_cc = os.path.join(self.package_folder, "bin", "musl-gcc") + " -static"
+            musl_cxx = musl_cc
+            musl_ld = musl_cc
+        elif self.settings.compiler == "clang":
+            musl_cc = os.path.join(self.package_folder, "bin", "musl-clang") + " -static"
+            musl_cxx = musl_cc
+            musl_ld = os.path.join(self.package_folder, "bin", "ld.musl-clang")
+
+        if musl_cc is not None:
+            self.output.info("Setting CC to '{}'".format(musl_cc))
+            self.env_info.CC = musl_cc
+        if musl_cxx is not None:
+            self.output.info("Setting CXX to '{}'".format(musl_cxx))
+            self.env_info.CXX = musl_cxx
+        if musl_ld is not None:
+            self.output.info("Setting LD to '{}'".format(musl_ld))
+            self.env_info.LD = musl_ld
```

Comment:
I would set `CC`,`CXX` and `LD` only when some option is enabled.
I think settings these environment variables unconditionally might break cross building.

Reply:
musl is an implementation of the C standard library. So we must link musl instead of the "original" standard library. musl provide wrapper that injects musl libc. However, these wrappers use hardcoded paths (See https://github.com/conan-io/conan-center-index/pull/2886#issuecomment-689894800). We must find a way to replace the "original" standard library with musl without hardcoded paths.

Reply:
Can this info be encoded on the `self.cpp_info` object?
It has `cflags`, `cxxflags`, `shared_link_flag`, `exe_link_flags`, ...